ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh

```diff
@@ -125,8 +125,26 @@ function is_openshift_version_gte() {
     printf '%s\n%s' "$1" "${DS_OPENSHIFT_VERSION}" | sort -C -V
 }
 
+function extract_oc_for_upgrade() {
+  # shellcheck disable=SC2087
+    ssh "${SSHOPTS[@]}" "root@${IP}" bash -x - << EOF
+EXTRACTDIR=\$(mktemp --tmpdir -d "installer-upgrade-XXXXXXXXXX")
+
+EXTRACTNEWOC="oc adm release extract --registry-config ${DS_WORKING_DIR}/pull_secret.json \
+   --command=oc --to "\${EXTRACTDIR}" $1"
+
+\$EXTRACTNEWOC
+
+sudo mv "\${EXTRACTDIR}/oc" /usr/local/bin
+
+echo "Extracted the new oc client on remote host:"
+oc version --client
```

Comment:
I can see that this extracts the new version of the "oc" client, I'm confused not about where it is actually used
The next part in this step is to run openshift-tests and then exit, does it need access to the new oc client? of so then maybe it needs to get copied to the local host... (or just extracted on the local host)